File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        Queue<Integer> pq = new PriorityQueue<>(scoville.length);
        for (int s : scoville) {
            pq.offer(s);
        }
        
        while (pq.peek() < K) {
            if (pq.size() == 1) {
                return -1;
            }
            int a = pq.poll();
            int b = pq.poll();
            pq.offer(a + b  * 2);
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    for (int s : scoville) {
        pq.push(s);
    }
    
    while (pq.top() < K) {
        if (pq.size() == 1) {
            return -1;
        }
        int a = pq.top(); pq.pop();
        int b = pq.top(); pq.pop();
        pq.push(a + b * 2);
        answer++;
    }
    
    return answer;
}